pkg/cluster/kubernetes/talos_managed.go

```diff
@@ -378,52 +371,23 @@ func getManifests(ctx context.Context, cluster UpgradeProvider) ([]*unstructured
 
 	defer cluster.Close() //nolint:errcheck
 
-	listClient, err := talosclient.Resources.List(ctx, k8s.ControlPlaneNamespaceName, k8s.ManifestType)
+	md, _ := metadata.FromOutgoingContext(ctx)
+	if nodes := md["nodes"]; len(nodes) > 0 {
```

Comment:
Not for this PR maybe, but would be cool to use some constants for metadata fields

Reply:
yeah, or even make this a generic loop, un-rolling `nodes` into `node` and a for loop (?)